### DISCUSSION TITLE
Is `fetcher` called on server side?

### DISCUSSION THREAD

**Discussion by dora-gt** [vercel/swr#2481] [Q&A]
Hi,

I'm wondering if `fetcher` is called on the server side or not.

I wrote some experimental code to confirm whether `fetcher` is called on the server side and found that it is called even on the server side in "some" context like the `suspense` option is true. I know in most cases, data should be fetched in the `getServerSideProps` section but `useSWR` seems to be also called in the server side context. Why I'd want to know that is that it critically affects `key`. If the `useSWR` is called on the server side, the key should contain a per-user one, shouldn't it? Otherwise, the other users' data could be exposed.

Could anyone help me?

**Comment by dora-gt** [comment]
ping

**Comment by icyJoseph** [comment]
Since React 18, it does get called during SSR... if you use suspense set to true. However, you should also see an error on your console, saying something like, "Fallback data must be provided", IIRC. 

I am not sure about what issue you see?

> Why I'd want to know that is that it critically affects key. If the useSWR is called on the server side, the key should contain a per-user one, shouldn't it? Otherwise, the other users' data could be exposed.

Like, if you do pre-rendering on the client, then how would you ever know which user is affected? it'd affect them all... 

And likewise, during GSSP-on-demand rendering, if its awaited on the server, or rendered client side, who would that expose user data?

Could you explain your idea/worry further?

**Reply by dora-gt** [reply]
Thank you for the reply!

> I am not sure about what issue you see?

If fetcher is called on servers, it must contain per-user key. I don't think it is an issue but we definitely should be aware of it because it may leak the other users' data if we don't specify per-user key as it caches.

Assume that we have `/api/getMyItems` API which returns the user's shopping items. We call `useSWR` with a fetcher which calls `/api/getMyItems`. Now user A visits the page and data of `/api/getMyItems` for user A is cached. Then user B visits. The data of user A should be rendered.

It could be avoided using a per-user key like `['MY_ITEMS', user_id]`.

What do you think?
Thanks!

**Reply by icyJoseph** [reply]
You know I hadn't checked out version 2.x.x until yesterday hehe ~ in this version:

> When using suspense mode on the server-side (including pre-rendering in Next.js), it's required to provide the initial data via fallbackData or fallback. This means that you can't use Suspense to fetch data on the server side, but either doing fully client-side data fetching, or fetch the data via the framework level data fetching method

https://swr.vercel.app/docs/suspense#server-side-rendering

They changed it so that this sort of worry you have is not possible -

And still suspense seems to be, not recommended.

**Reply by dora-gt** [reply]
So, you mean `fetcher` is NOT called on server side, is that right?

Thanks!
